src/config.rs

```diff
@@ -32,6 +33,20 @@ pub struct SubsetConfig {
     pub base_url: Option<String>,
 }
 
+/// A command to run during the build/publish process. Given as an array of arguments, with the first argument being
+/// the path to the program. Arguments are not processed by a shell; they are passed directly to the program.
+#[derive(Deserialize, Debug, Default, Clone)]
+pub struct ConfigHook(Option<Vec<String>>);
+
+/// Defines a set of hook commands to run at certain points in the build/publish process.
+#[derive(Deserialize, Debug, Default, Clone)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct ConfigHooks {
+    /// Runs during publish jobs before the build is imported to the main repository. The hook is allowed to edit the
+    /// repository. The current directory is set to the build directory.
+    pub publish: ConfigHook,
```

Comment:
The way this is written, it's valid for the configuration to contain `publish: null` which seems odd/broken.  I'd far rather see this be `pub publish: Option<ConfigHook>` instead.